akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala

```diff
@@ -65,9 +73,25 @@ object Route {
                                 exceptionHandler: ExceptionHandler         = null): Flow[HttpRequest, HttpResponse, NotUsed] =
     Flow[HttpRequest].mapAsync(1)(asyncHandler(route))
 
+  def asyncHandler(route: Route, routingSettings: RoutingSettings, parserSettings: ParserSettings)(implicit system: ActorSystem): HttpRequest => Future[HttpResponse] = {
+    val routingLog = RoutingLog(system.log)
+    implicit val executionContext = system.dispatcher
+    implicit val materializer = Materializer.matFromSystem(new ClassicActorSystemProvider {
```

Comment:
Isn't this an elaborate way of saying `SystemMaterializer(system).materializer`?